diffusion/evaluation/clean_fid_eval.py

```diff
@@ -0,0 +1,237 @@
+        size (int): The size of the images to generate. Default: ``256``.
+        batch_size (int): The per-device batch size to use for evaluation. Default: ``16``.
+        loggers (List[LoggerDestination], optional): The loggers to use for logging results. Default: ``None``.
+        seed (int): The seed to use for evaluation. Default: ``17``.
+        output_dir (str): The directory to save results to. Default: ``/tmp/``.
+        num_samples (int, optional): The maximum number of samples to generate. Depending on batch size, actual
+            number may be slightly higher. If not specified, all the samples in the dataloader will be used.
+            Default: ``None``.
+        precision (str): The precision to use for evaluation. Default: ``'amp_fp16'``.
+        prompts (List[str], optional): The prompts to use for image visualtization.
+            Default: ``["A shiba inu wearing a blue sweater]``.
+
+    """
+
+    def __init__(self,
+                 model: ComposerModel,
+                 eval_dataloader: DataLoader,
+                 clip_metric: CLIPScore,
+                 load_path: Optional[str] = None,
+                 guidance_scales: Optional[List[float]] = None,
+                 size: int = 256,
+                 batch_size: int = 16,
+                 loggers: Optional[List[LoggerDestination]] = None,
+                 seed: int = 17,
+                 output_dir: str = '/tmp/',
+                 num_samples: Optional[int] = None,
+                 precision: str = 'amp_fp16',
+                 prompts: Optional[List[str]] = None):
+        self.model = model
+        self.tokenizer: PreTrainedTokenizerBase = model.tokenizer
+        self.eval_dataloader = eval_dataloader
+        self.clip_metric = clip_metric
+        self.load_path = load_path
+        self.guidance_scales = guidance_scales if guidance_scales is not None else [1.0]
+        self.size = size
+        self.batch_size = batch_size
+        self.loggers = loggers
+        self.seed = seed
+        self.output_dir = output_dir
+        self.num_samples = num_samples if num_samples is not None else float('inf')
+        self.precision = precision
+        self.prompts = prompts if prompts is not None else ['A shiba inu wearing a blue sweater']
+
+        # Init loggers
+        if self.loggers and dist.get_local_rank() == 0:
+            for logger in self.loggers:
+                if isinstance(logger, WandBLogger):
+                    wandb.init(**logger._init_kwargs)
+
+        # Load the model
+        Trainer(model=self.model,
+                load_path=self.load_path,
+                load_weights_only=True,
+                eval_dataloader=self.eval_dataloader)
+
+        # Move CLIP metric to device
+        self.device = dist.get_local_rank()
+        self.clip_metric = self.clip_metric.to(self.device)
+
+        # Predownload the CLIP model for computing clip-fid
+        _, _ = clip.load('ViT-B/32', device=self.device)
```

Comment:
Is this required?

Reply:
This isn't strictly required, but it helps to do this download up front. Otherwise the clip model gets downloaded the first time the clip-fid metric is computed. This adds an extra import here, but not an extra dependency since the `clip` package is already required to compute `clean-fid`'s clip-fid metric.

Reply:
If the clip package is installed with `clean-fid`, do we need to include it in the setup.py? i.e. `clip@git+https://github.com/openai/CLIP.git`